acceptance-tests/config.go

```diff
@@ -0,0 +1,141 @@
+	BoshDirectorKey  string `json:"boshDirectorKey"`
+	BoshDirectorCA   string `json:"boshDirectorCA"`
+	BoshClient       string `json:"boshClient"`
+	BoshClientSecret string `json:"boshClientSecret"`
+	BoshEnvironment  string `json:"boshEnvironment"`
+	BoshPath         string `json:"boshPath"`
+	BaseManifestPath string `json:"baseManifestPath"`
+	HomePath         string `json:"homePath"`
+}
+
+func loadConfig() (Config, error) {
+	releaseRepoPath, err := getEnvOrFail("REPO_ROOT")
+	if err != nil {
+		return Config{}, err
+	}
+
+	releaseVersion, err := getEnvOrFail("RELEASE_VERSION")
+	if err != nil {
+		return Config{}, err
+	}
+
+	boshDirectorCA, err := getEnvOrFail("BOSH_DIRECTOR_CA")
+	if err != nil {
+		return Config{}, err
+	}
+
+	boshDirectorCert, err := getEnvOrFail("BOSH_DIRECTOR_CERT")
+	if err != nil {
+		return Config{}, err
+	}
+
+	boshDirectorKey, err := getEnvOrFail("BOSH_DIRECTOR_KEY")
+	if err != nil {
+		return Config{}, err
+	}
+
+	boshClient, err := getEnvOrFail("BOSH_CLIENT")
+	if err != nil {
+		return Config{}, err
+	}
+
+	boshClientSecret, err := getEnvOrFail("BOSH_CLIENT_SECRET")
+	if err != nil {
+		return Config{}, err
+	}
+
+	boshDirectorAPI, err := getEnvOrFail("BOSH_DIRECTOR_API")
+	if err != nil {
+		return Config{}, err
+	}
+
+	boshEnvironment, err := getEnvOrFail("BOSH_ENVIRONMENT")
+	if err != nil {
+		return Config{}, err
+	}
+
+	boshPath, err := getEnvOrFail("BOSH_PATH")
+	if err != nil {
+		return Config{}, err
+	}
+
+	baseManifestPath, err := getEnvOrFail("BASE_MANIFEST_PATH")
+	if err != nil {
+		return Config{}, err
+	}
+
+	// BOSH commands require HOME is set
+	homePath, err := getEnvOrFail("HOME")
+	if err != nil {
+		return Config{}, err
+	}
+
+	return Config{
+		ReleaseRepoPath:  releaseRepoPath,
+		ReleaseVersion:   releaseVersion,
+		BoshDirectorAPI:  boshDirectorAPI,
+		BoshDirectorCert: boshDirectorCert,
+		BoshDirectorKey:  boshDirectorKey,
+		BoshDirectorCA:   boshDirectorCA,
+		BoshClient:       boshClient,
+		BoshClientSecret: boshClientSecret,
+		BoshEnvironment:  boshEnvironment,
+		BoshPath:         boshPath,
+		BaseManifestPath: baseManifestPath,
+		HomePath:         homePath,
+	}, nil
+}
+
+func (config *Config) boshCmd(boshDeployment string, args ...string) *exec.Cmd {
+	cmd := exec.Command(config.BoshPath, append([]string{"--tty", "--no-color"}, args...)...)
+	cmd.Env = []string{
+		fmt.Sprintf("BOSH_DIRECTOR_IP=%s", config.BoshDirectorAPI),
+		fmt.Sprintf("BOSH_DIRECTOR_CA=%s", config.BoshDirectorCA),
+		fmt.Sprintf("BOSH_DIRECTOR_CERT=%s", config.BoshDirectorCert),
+		fmt.Sprintf("BOSH_CLIENT=%s", config.BoshClient),
+		fmt.Sprintf("BOSH_CLIENT_SECRET=%s", config.BoshClientSecret),
+		fmt.Sprintf("BOSH_ENVIRONMENT=%s", config.BoshEnvironment),
+		fmt.Sprintf("HOME=%s", config.HomePath),
+		fmt.Sprintf("BOSH_DEPLOYMENT=%s", boshDeployment),
+		"BOSH_NON_INTERACTIVE=true",
+	}
+
+	return cmd
+}
+
+func (config *Config) boshInteractiveCmd(args ...string) *exec.Cmd {
```

Comment:
We can have a boshBasicCmd like 
```
func (config *Config) boshBasicCmd(args ...string) *exec.Cmd {
	cmd := exec.Command(config.BoshPath, append([]string{"--tty", "--no-color"}, args...)...)
	cmd.Env = []string{
		fmt.Sprintf("BOSH_DIRECTOR_IP=%s", config.BoshDirectorAPI),
		fmt.Sprintf("BOSH_DIRECTOR_CA=%s", config.BoshDirectorCA),
		fmt.Sprintf("BOSH_DIRECTOR_CERT=%s", config.BoshDirectorCert),
		fmt.Sprintf("BOSH_ENVIRONMENT=%s", config.BoshEnvironment),
		fmt.Sprintf("HOME=%s", config.HomePath),
	}

	return cmd
}
```

And then reuse the function : 
```
func (config *Config) boshCmd(boshDeployment string, args ...string) *exec.Cmd {
	cmd := config.boshBasicCmd(args)
	cmd.Env = append(cmd.Env, 
		fmt.Sprintf("BOSH_CLIENT=%s", config.BoshClient),
		fmt.Sprintf("BOSH_CLIENT_SECRET=%s", config.BoshClientSecret),
		fmt.Sprintf("BOSH_DEPLOYMENT=%s", boshDeployment),
		"BOSH_NON_INTERACTIVE=true")

	return cmd
}

func (config *Config) boshInteractiveCmd(args ...string) *exec.Cmd {
	return config.boshBasicCmd(args)
}
```

Reply:
yes, but it would change the meaning. the point was to have one interactive command that requires user input and one non-interactive command that expects user creds as variables. Your `boshBasicCmd` is the same as `boshInteractiveCmd`.

So if at all, you would want to have `boshInteractiveCmd` be the "basic" one and let `boshCmd` add to it, but that also feels kinda off.

Reply:
I did not want to rename the function `boshInteractiveCmd`, as you are right it will change the meaning of the function. But I wanted to have the same coding on one place.